File: ui/admin/src/types/page.ts
/**
 * 独立页面相关类型定义
 */

/** 页面状态枚举 */
export type PageStatus = 'published' | 'draft'

/** 独立页面数据结构 */
export interface Page {
  id: string
  title: string
  slug: string
  status: PageStatus
  /** 排序优先级，数值越小越靠前 */
  sortOrder: number
  /** 是否在前台导航栏中显示 */
  showInNav: boolean
  createdAt: string
  updatedAt: string
  publishedAt: string | null
}

/** 独立页面详情（含正文） */
export interface PageDetail extends Page {
  content: string
}

/** 独立页面表单数据 */
export interface PageFormData {
  title: string
  slug: string
  content: string
  status: PageStatus
  sortOrder: number
  showInNav: boolean
}

File: internal/model/page.go
package model

import "time"

const (
	PageStatusDraft     = "draft"
	PageStatusPublished = "published"
)

// Page 独立页面模型
type Page struct {
	BaseModel
	Title           string     `gorm:"size:255;not null" json:"title"`
	Slug            string     `gorm:"size:255;not null;uniqueIndex" json:"slug"`
	ContentMarkdown string     `gorm:"type:text;not null" json:"content_markdown"`
	ContentHTML     string     `gorm:"type:text;not null" json:"content_html"`
	Status          string     `gorm:"size:32;not null;default:draft;index" json:"status"`
	SortOrder       int        `gorm:"not null;default:0;index" json:"sort_order"`
	ShowInNav       bool       `gorm:"not null;default:false" json:"show_in_nav"`
	PublishedAt     *time.Time `json:"published_at"`
	Template        string     `gorm:"size:64;not null;default:default" json:"template"` // 模板名，对应 templates/pages/{template}.html
	Config          string     `gorm:"type:text" json:"config"`                          // 模板参数，JSON 格式
}
